src/cisco_gnmi/nx.py

```diff
@@ -53,11 +57,296 @@ class NXClient(Client):
+        ----------
+        xpaths : iterable of str
+            XPaths to specify to be deleted.
+            If prefix is specified these strings are assumed to be the suffixes.
+        prefix : str
+            The XPath prefix to apply to all XPaths for deletion.
+
+        Returns
+        -------
+        set()
+        """
+        if isinstance(xpaths, string_types):
+            xpaths = [xpaths]
+        paths = []
+        for xpath in xpaths:
+            if prefix:
+                if prefix.endswith("/") and xpath.startswith("/"):
+                    xpath = "{prefix}{xpath}".format(
+                        prefix=prefix[:-1], xpath=xpath[1:]
+                    )
+                elif prefix.endswith("/") or xpath.startswith("/"):
+                    xpath = "{prefix}{xpath}".format(prefix=prefix, xpath=xpath)
+                else:
+                    xpath = "{prefix}/{xpath}".format(prefix=prefix, xpath=xpath)
+            paths.append(self.parse_xpath_to_gnmi_path(xpath))
+        return self.set(deletes=paths)
+
+    def check_configs(self, configs):
+        if isinstance(configs, string_types):
+            logger.debug("Handling as JSON string.")
+            try:
+                configs = json.loads(configs)
+            except:
+                raise Exception("{0}\n is invalid JSON!".format(configs))
+            configs = [configs]
+        elif isinstance(configs, dict):
+            logger.debug("Handling already serialized JSON object.")
+            configs = [configs]
+        elif not isinstance(configs, (list, set)):
+            raise Exception(
+                "{0} must be an iterable of configs!".format(str(configs))
+            )
+        return configs
+
+    def create_updates(self, configs, origin, json_ietf=False):
+        """Check configs, and construct "Update" messages.
 
-    def set(self, *args, **kwargs):
-        raise NotImplementedError("Set not yet supported on NX-OS!")
+        Parameters
+        ----------
+        configs: dict of <xpath>: <dict val for JSON>
+        origin: str [DME, device, openconfig]
+        json_ietf: bool encoding type for Update val (default False)
+
+        Returns
+        -------
+        List of Update messages with val populated.
+
+        If a set of configs contain a common Xpath, the Update must contain
+        a consolidation of xpath/values for 2 reasons:
+
+        1. Devices may have a restriction on how many Update messages it will
+           accept at once.
+        2. Some xpath/values are required to be set in same Update because of
+           dependencies like leafrefs, mandatory settings, and if/when/musts.
+        """
+        if not configs:
+            return []
+        configs = self.check_configs(configs)
+
+        xpaths = []
+        updates = []
+        for config in configs:
+            xpath = next(iter(config.keys()))
+            xpaths.append(xpath)
+        common_xpath = os.path.commonprefix(xpaths)
```

Comment:
Evaluate if `commonprefix` is good beyond `os.path` understanding of path.

Reply:
It is:
```
>>> xp3='/Users/miott/pyats/develop[my="me"]/yang/connector/src'
>>> xp4='/Users/miott/pyats/develop[my="me"]/yang/connector/other/stuff'
>>> os.path.commonprefix([xp3,xp4])
'/Users/miott/pyats/develop[my="me"]/yang/connector/'
>>> 
```